js/fullscreenpost.js

```diff
@@ -0,0 +1,135 @@
+const commentsList = fullscreenPicture.querySelector('.social__comments');
+const fullscreenPictureShowCommentsCount = fullscreenPicture.querySelector('.social__comment-shown-count');
+const fullscreenPictureTotalCommentCount = fullscreenPicture.querySelector('.social__comment-total-count');
+const fullscreenPictureLoadMoreComment = fullscreenPicture.querySelector('.comments-loader');
+
+const SHOW_COMMENTS_COUNT = 5;
+
+const fullscreenPictureLikesCount = fullscreenPicture.querySelector('.likes-count');
+const postsList = document.querySelector('.pictures');
+
+//объявляем функции
+let closeFullscreenPicture = function(){};
+let loadMoreComment = function(){};
+
+// закрываем окно с большой картинкой клавишей esc
+const onDocumentKeydown = (evt) => {
+  if (isEscapeKey(evt)) {
+    evt.preventDefault();
+    closeFullscreenPicture();
+  }
+};
+
+// закрываем окно с большой картинкой крестиком
+closeFullscreenPicture = () => {
+  fullscreenPicture.classList.add('hidden');
+  document.body.classList.remove('modal-open');
+  commentsList.innerHTML = '';
+  document.removeEventListener('keydown', onDocumentKeydown);
+  fullscreenPictureCloseButton.removeEventListener('click', closeFullscreenPicture);
+  fullscreenPictureLoadMoreComment.removeEventListener('click', loadMoreComment);
+
+};
+
+// открываем окно с большой картинкой
+const openFullscreenPicture = () => {
+  fullscreenPicture.classList.remove('hidden');
+  document.body.classList.add('modal-open');
+  document.addEventListener('keydown', onDocumentKeydown);
+  fullscreenPictureCloseButton.addEventListener('click', closeFullscreenPicture);
+};
+
+// отрисовывем комментарии
+const renderComments = (comments) => {
+
+  const commentFragment = document.createDocumentFragment();
+  commentsList.innerHTML = '';
+
+  comments.forEach((comment, index) => {
+    const commentItem = commentTemplate.cloneNode(true);
+
+    commentItem.querySelector('.social__picture').src = comment.avatar;
+    commentItem.querySelector('.social__picture').alt = comment.name;
+    commentItem.querySelector('.social__text').textContent = comment.message;
+
+    if (index > SHOW_COMMENTS_COUNT - 1) {
+      commentItem.classList.add('hidden');//скрываем комментарии кроме 5 первых
+    }
+    commentFragment.appendChild(commentItem);
+  });
+
+  commentsList.appendChild(commentFragment);
+};
+
+//обработчик кнопки загрузки комментариев
+loadMoreComment = () => {
+  const commentHiddenArray = commentsList.querySelectorAll('.social__comment.hidden');//находим массив скрытых комментариев
+  let countD = 0;
+
+  if(commentHiddenArray.length >= SHOW_COMMENTS_COUNT){
+    for (let i = 0; i < SHOW_COMMENTS_COUNT; i++){
+      commentHiddenArray[i].classList.remove('hidden');
+      countD++;
+    }
+  } else {
+    for (let i = 0; i < commentHiddenArray.length; i++){
+      commentHiddenArray[i].classList.remove('hidden');
+      countD++;
+    }
```

Comment:
for (let i = 0; i < Math.min(commentHiddenArray.length, SHOW_COMMENTS_COUNT); i++) {}